core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/AbstractDynamicRegistry.java

```diff
@@ -40,7 +39,7 @@
  * Base implementation for {@link org.apache.camel.spi.EndpointRegistry},
  * {@link org.apache.camel.spi.TransformerRegistry}, and {@link org.apache.camel.spi.ValidatorRegistry}.
  */
-public class AbstractDynamicRegistry<K, V> extends AbstractMap<K, V> implements StaticService {
+public class AbstractDynamicRegistry<K, V> extends ConcurrentHashMap<K, V> implements StaticService {
```

Comment:
TBH, I kinda dislike the fact that it extends a core Java class. I wonder if we should, maybe, re-evaluate how we do this to avoid tightly coupling here.

Reply:
I agree with you. This PR is solving an issue. I think we can fix this issue with this code and open an improvement issue to review this point.

Reply:
To be clear: I am not saying you should do this now. I'm just thinking out loud ...